### DISCUSSION TITLE
Artifacts in CI

### DISCUSSION THREAD

**Discussion by SimonasB88** [cypress-io/cypress#15134] [CI setup] [answered]
Hey guys,

wanted to ask, if there is possible to configure the Actions for CI test like that and succeed the test execution: 

1. Downloading the artifact (after the PR) to “cypress/fixtures” path 
2. Uploading the artifact into the application using the Cypress file upload plugin

Thank you!

**Comment by bahmutov** [comment]
Can you provide more explanation, please? What artifacts? Download after PR? Do you mean recording test results and videos and screenshots to Cypress Dashboard and then getting them back?

**Reply by SimonasB88** [reply]
Hey @bahmutov thank you for your answer!

I mean, there is a Payment plugin that must be tested in production server. Every time in PR there is a job workflow:
![image](https://user-images.githubusercontent.com/29095235/108475101-f0792a00-7298-11eb-950b-404798ea52ac.png)

Then job is uploading the artifact to "fixtures" path in the Github
![image](https://user-images.githubusercontent.com/29095235/108475240-1dc5d800-7299-11eb-9cbb-f427ae7348e7.png)

And then, I would like to upload that artifact to the production server, in this case it is plugin .zip, and to run the test in pipeline on production server.
Cypress File Upload works ok for that, but the test can't get that uploaded artifact in "fixtures" folder.

Is it possible to test like that?

Thanks!

**Comment by bahmutov** [comment]
I’m still very confused by what you are trying to do. Are you trying to upload a file during cypress test?

**Reply by SimonasB88** [reply]
@bahmutov yes, upload plugin .zip file, into the ecommerce platform, but I am trying to do this on pipeline. 

Looking forward to hearing from you!

**Comment by bahmutov** [comment, marked as answer]
Can you show the spec file that you have so far and show screenshots of what happens?

**Reply by SimonasB88** [reply]
Sure,
The .js test file:
```
cy.get('#subtab-AdminParentModulesSf > :nth-child(1)').click()
      cy.get('#subtab-AdminModulesSf > .link').click()
      cy.get('#page-header-desc-configuration-add_module').click()
      cy.get('.module-import-start').attachFile('fixture.zip', { subjectType: 'drag-n-drop' }).wait(15000)
      cy.get('.module-import-success-msg').should('be.visible').as('Module installed!')
```
The pipeline executes like that:

![image](https://user-images.githubusercontent.com/29095235/108699886-82d63380-750e-11eb-98bd-7681a65c6c24.png)

And cypress can not find the "fixture.zip":

![image](https://user-images.githubusercontent.com/29095235/108700321-17d92c80-750f-11eb-9998-6021a7e9bd0e.png)

It is seen in the run although:

![image](https://user-images.githubusercontent.com/29095235/108700367-27587580-750f-11eb-8f93-faf29c09dba9.png)

If i add a plugin zip in fixtures folder instead of "fixtures.zip", the outcome is the same, pipeline can not find the .zip. 

Looking forward to hearing from you!

**Reply by SimonasB88** [reply]
Hey @bahmutov any thoughts in this workflow?

Thanks!

**Reply by bahmutov** [reply]
I still don't understand this. Are you testing GitHub Actions? The previous download actions placed a zip into cypress/fixtures folder? the command `.attachFile` is not a built-in command, it probably comes from https://github.com/abramenal/cypress-file-upload plugin - does it expect the file name relative to the fixtures folder? Does it work when you simply have a zip file in the fixtures folder without download plugin?

**Reply by SimonasB88** [reply]
@bahmutov hey! Thanks for your questions, sorry, if some misunderstandings are appearing, I am not so experienced so far.
I am trying to make automated steps, that once the developer makes PR, the Github Actions artifact (it becomes plugin zip) is being uploaded to the ecommerce for testing.

Yes, if I place zip, to the fixtures folder in repo, that is ok, the zip uploads to the ecommerce
<img width="839" alt="image" src="https://user-images.githubusercontent.com/29095235/109625890-88ec9580-7b48-11eb-875d-ee4402561bf8.png">

Regarding the https://github.com/abramenal/cypress-file-upload - I didn't provide the correct path to fixtures folder maybe?
How can I correct the code?

```
cy.get('#subtab-AdminParentModulesSf > :nth-child(1)').click()
      cy.get('#subtab-AdminModulesSf > .link').click()
      cy.get('#page-header-desc-configuration-add_module').click()
      cy.get('.module-import-start').attachFile('fixture.zip', { subjectType: 'drag-n-drop' }).wait(15000)
      cy.get('.module-import-success-msg').should('be.visible').as('Module installed!')
```
I am so thankful for your suggestions.

Best Regards.

**Reply by bahmutov** [reply]
Can you set up a public repo with the GitHub workflow file showing the problem? Have you checked if the files are present before the Cypress test run?

**Reply by SimonasB88** [reply]
@bahmutov 
Today checked the night test run with fake "fixture.zip", the pipeline succeeded:
![image](https://user-images.githubusercontent.com/29095235/109776035-d92c2c00-7c0a-11eb-9ef5-4e3a2dd80f4e.png)
And file in repo:
![image](https://user-images.githubusercontent.com/29095235/109776207-04168000-7c0b-11eb-94e2-4405c10afd38.png)

I am very sorry, I will have to make another repo then, because due to privacy of the company, I can not do this now... I will try to do another public repo.
Wondering, if you had some examples of testing the github artifact, that becomes a ecommerce plugin zip?

Thanks!

**Reply by bahmutov** [reply]
I have examples of using the upload and download artifacts on GitHub, but no examples of uploading the zip in Cypress test. I would think this is a bad strategy, since you don't control the file. I would suggest using a zip file you have already in the fixtures folder as part of the repo. Also, I think the zip download is NOT downloading where you expect it to be, use `ls` command to print the downloaded file path. 

I don't think to create a public repo with just this workflow is difficult, otherwise, I am suggesting things mindly.

**Reply by SimonasB88** [reply]
@bahmutov Thanks Gleb for your suggestions, I guess lessons learned. I will use your suggestions. Thank you for your thoughts and for t-shirt also :)

Best Regards.
